format/Message.fbs

```diff
@@ -42,6 +42,38 @@ struct FieldNode {
   null_count: long;
 }
 
+enum CompressionType:byte {
+  // LZ4 frame format, for portability, as provided by lz4frame.h or wrappers
+  // thereof. Not to be confused with "raw" (also called "block") format
+  // provided by lz4.h
+  LZ4_FRAME,
+
+  // Zstandard
+  ZSTD
+}
+
+/// Provided for forward compatibility in case we need to support different
+/// strategies in the future
+enum RecordBatchCompressionMethod:byte {
+  /// Each constituent buffer is first compressed with the indicated
+  /// compressor, and then written with the uncompressed length in the first 8
+  /// bytes as a 64-bit signed integer followed by the compressed buffer bytes
+  /// (and then padding as required by the protocol). The uncompressed length
+  /// may be set to -1 to indicate that the data that follows is not
+  /// compressed, which can be useful for cases where compression does not
+  /// yield appreciable savings.
+  BUFFER
+}
+
+/// Optional compression for record batches
+table RecordBatchCompression {
```

Comment:
aside: given this table to you still think things like RLE belong in their own message type?

Reply:
I think the logical compression should be hinted in FieldNode.

Reply:
I view this whole effort as being separate from in-memory encoding / compression and sparseness. I still think we need an `EncodedRecordBatch` type (called `SparseRecordBatch` in your patch #4815). It seems to me that we will want a separate set of metadata structures for encoding those details (permitting encoding types to come from a Union for better extensibility)

If it helps let's come up with a better name for this table here so it doesn't confuse people. "IpcBodyCompression"?